src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderGlobal_BiggerRenderChunk_Common.java

```diff
@@ -0,0 +1,230 @@
+    }
+
+    @ModifyConstant(method = "renderWorldBorder", constant = {
+            @Constant(doubleValue = 0.0D),
+            @Constant(doubleValue = 256.0D)
+    }, slice = @Slice(from = @At(value = "HEAD"), to = @At(value = "INVOKE", target = "Lnet/minecraft/client/renderer/Tessellator;draw()V")), require = 2)
+    private double renderWorldBorder_getRenderHeight(double original, Entity entity, float partialTicks) {
+        return original == 0.0D ? entity.posY - 128 : entity.posY + 128;
+    }
+
+    @Redirect(method = "setupTerrain", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/renderer/chunk/RenderChunk;needsUpdate()Z"))
+    public boolean doChunkNeedUpdate(RenderChunk renderChunk) {
+        int maxPos = RenderVariables.getRenderChunkMaxPos();
+        return renderChunk.needsUpdate() && world.isAreaLoaded(renderChunk.getPosition(),
+                renderChunk.getPosition().add(maxPos, maxPos, maxPos));
+    }
+
+    @ModifyConstant(method = "setupTerrain", constant = @Constant(doubleValue = 16.0D))
+    public double onSetupTerrain1(double oldValue) {
+        return Double.MAX_VALUE;
+    }
+
+    @ModifyConstant(method = "setupTerrain", constant = @Constant(intValue = 16))
+    public int onSetupTerrain2(int oldValue) {
+        return RenderVariables.getRenderChunkSize();
+    }
+    
+    @Inject(method = "setupTerrain", at = @At(value = "HEAD"), cancellable = false)
+    public void reloadRendersIfNecessary(Entity viewEntity, double partialTicks, ICamera camera, int frameCount, boolean playerSpectator,
+            CallbackInfo ci) {
+        if (this.renderChunkSizeBit != RenderVariables.getRenderChunkBit()) {
+            this.loadRenderers();
+            this.renderDistanceChunks = this.mc.gameSettings.renderDistanceChunks;
+            this.renderChunkSizeBit = RenderVariables.getRenderChunkBit();
+        }
+    }
+
+    /**
+     * @author Foghrye4
+     * @reason function is short, simple and does not allow to use bigger render chunks.
+     */
+    @Overwrite
+    private Set<EnumFacing> getVisibleFacings(BlockPos pos) {
+        VisGraph visgraph = new VisGraph();
+        int mask = RenderVariables.getRenderStartPosMask();
+        int renderChunkCubeSize = RenderVariables.getRenderChunkSize() / Cube.SIZE;
+        int bx = pos.getX() & mask;
+        int by = pos.getY() & mask;
+        int bz = pos.getZ() & mask;
+        int cx0 = Coords.blockToCube(bx);
+        int cy0 = Coords.blockToCube(by);
+        int cz0 = Coords.blockToCube(bz);
+        ICubicWorldClient cworld = (ICubicWorldClient) world;
+        if (cworld.isCubicWorld()) {
+            CubeProviderClient cubeProvider = cworld.getCubeCache();
+            for (int cx = cx0; cx < cx0 + renderChunkCubeSize; cx++)
+                for (int cy = cy0; cy < cy0 + renderChunkCubeSize; cy++)
+                    for (int cz = cz0; cz < cz0 + renderChunkCubeSize; cz++) {
```

Comment:
This kind of code everywhere is exactly why I planned to make virtual 32x32x32 cube class. That could be a separate PR. It would wrap these kinds of loops and handle it by itself. If it also implemented ICube interface with `getSize` method, you could still dynamicall switch between them without duplicating all the rendering code.